lib/common/util.js

```diff
@@ -545,3 +545,43 @@ function normalizeArguments(globalContext, localConfig, options) {
 }
 
 util.normalizeArguments = normalizeArguments;
+
+/**
+ * Throttle requests according to a `maxApiCalls` limit.
+ *
+ * @param {function} makeRequestFn - The function that will be called.
+ * @param {object=} options - Configuration object.
+ * @param {number} options.maxApiCalls - The maximum number of API calls to
+ *     make.
+ */
+function iterator(makeRequestFn, options) {
+  var stream = through.obj();
+
+  var requestsMade = 0;
+  var requestsToMake = -1;
+
+  options = options || {};
```

Comment:
I did initially try to fit that in. My issue was that it took away too much control from the originating method. Having it called automatically is nice, but (when I was thinking along the lines of this being a generator), it blanketed the call, and made the next "iterator.makeRequest()" feel unexpected.

There's also the issue of where the original options the function is called with come from. If util.createLimiter handles it, we would have to provide the initial objects to `createLimiter`.

I just made a change that made the stream a "stream-events"-ified one inside of `util.createLimiter` itself. Now we don't need to worry about that logic wherever we use a limiter.